src/fides/cli/options.py

```diff
@@ -38,6 +38,18 @@ def resource_type_argument(command: Callable) -> Callable:
     return command
 
 
+def resource_type_option(command: Callable) -> Callable:
+    "Add the resource_type option."
+    command = click.option(
+        "--resource-type",
```

Comment:
I _really_ prefer this as an argument (e.g. `fides pull dataset {key}`) than as an option (`--resource-type`). It just feels out of place to me?

Is there a particular reason you didn't use the argument type?

Reply:
In my mind (and the default behavior for click) an "argument" like that is a requirement, non optional, whereas an option is optional. It looks like we could [override the behavior](https://stackoverflow.com/questions/44056153/optional-argument-in-command-with-click) and write a custom class to make optional click arguments if you feel strongly enough

Reply:
This is all a consideration because `fides push` by default pushes everything. We could also split out this functionality into a different command

Reply:
I'd avoid trying to hack the Click internals at all for this, they are well considered to make parsing a terminal command unambiguously.

That said... can you do this without achieve this by just having the "push" command group have a few subcommands with names like:
- name=system
- name=dataset
- name=data_category
- name='' 

The empty string would effectively be the default sub-command. I think it might let you do that?

If it feels janky and weird, don't do it and follow the standard way for Click apps 👍